File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    
    count = 0 
    zero = 0
    while True:
        if s == '1':
            break
        zero = zero + s.count('0')
        s = s.replace('0', '')
        
        s = bin(len(s))[2:]
        count += 1
        
    answer = [count, zero]
    
    return answer

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    int countIteration = 0;
    int countZero = 0;
    
    while(s != "1") {
        int len = s.length();
        int cnt = 0;
        for(int i=0; i<len; i++) {
            if(s[i] == '0') {
                cnt++;
            }
        }
        countZero += cnt;
        len -= cnt;
        s = "";
        while(len > 0) 
        {
            if(len % 2 == 1) {
                s += "1";
                cout << s << endl;
            } else {
                s += "0";
            }
            len /= 2;
        }
        countIteration++;
    }
    answer.push_back(countIteration);
    answer.push_back(countZero);
    
    return answer;
}